ls1intum/Artemis
# `Assessment`: Add tool tips for assessment notes

### Checklist
#### General


#### Server


#### Client


#### Changes affecting Programming Exercises
- [x] I tested **all** changes and their related features with **all** corresponding user types on ~~a test server~~ my local machine configured with **Gitlab** and **Jenkins**.


### Motivation and Context

https://github.com/ls1intum/Artemis/issues/1402#issuecomment-1607328925 explains why this feature is useful. While it is nice to have assessment notes https://github.com/ls1intum/Artemis/pull/8174, their full potential only shows once there is an easy way to see which assessments have such a note.

### Description

A new column is added to the scores/submissions table, which contains little speech bubbles indicating the presence of an internal assessment note for an entry. Moving your mouse over this icon shows the content of the assessment note for the respective submission.

### Steps for Testing

Prerequisites:

-  1 Instructor
-  1 Programming Exercise with manual assessment

1. Log in to Artemis
2. Follow the steps from https://github.com/ls1intum/Artemis/pull/8174 to create a programming exercise and add an assessment note
4. Go to the submissions overview of the exercise
5. There should be a speech bubble in the table for the exercise you just created a note for and moving your mouse over this icon should show your note

### Testserver States
> [!NOTE]
> These badges show the state of the test servers.
> Green = Currently available, Red = Currently locked

![](https://byob.yarr.is/ls1intum/Artemis/artemis-test1)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test2)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test3)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test4)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test5)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test6)

### Review Progress






#### Performance Review
#### Code Review
#### Manual Tests

### Test Coverage
Only changes in SQL code on the server side.

### Screenshots
#### Showing Tooltip
![screenshot tooltips for assessment notes](https://github.com/ls1intum/Artemis/assets/73783062/326cac9f-8747-4e6b-bc15-001ffcc35dc4)

#### Sorting column
![screenshot sorting column](https://github.com/ls1intum/Artemis/assets/73783062/4b62f8e2-2da3-45ed-9cb3-b7dfb04f507a)
